runelite-client/src/main/java/net/runelite/client/plugins/zalcano/ZalcanoPlugin.java

```diff
@@ -0,0 +1,233 @@
+ *  *    list of conditions and the following disclaimer.
+ *  * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *  *    this list of conditions and the following disclaimer in the documentation
+ *  *    and/or other materials provided with the distribution.
+ *  * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ *  * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ *  * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ *  * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ *  * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ *  * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ *  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package net.runelite.client.plugins.zalcano;
+
+import com.google.inject.Binder;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.NPC;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.NpcDespawned;
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.api.kit.KitType;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.EventBus;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.PluginType;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Zalcano",
+	description = "Highlights Zalcano AOEs and useful stuff",
+	tags = {"zalcano", "aoe", "prifddinas", "elf", "boss"},
+	type = PluginType.PVM
+)
+@Slf4j
+public class ZalcanoPlugin extends Plugin
+{
+
+	@Inject
+	Client client;
+
+	@Inject
+	ZalcanoConfig config;
+
+	@Inject
+	ZalcanoUtil util;
+
+	@Inject
+	ZalcanoOverlay overlay;
+
+	@Inject
+	ZalcanoStepsOverlay stepsOverlay;
+
+	@Inject
+	OverlayManager overlayManager;
+
+	@Inject
+	private EventBus eventBus;
+
+	protected static final String mine = "MINE";
+	protected static final String warning = "GET BACK";
+
+	@Setter
+	@Getter
+	private NPC zalcano;
+
+	@Getter
+	private NPC golem;
+
+	@Getter
+	@Setter
+	private Step step;
+
+	private int ores = 0;
+
+	@Provides
+	ZalcanoConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(ZalcanoConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		eventBus.subscribe(GameTick.class, "regionchecker", this::onGameTickCheckRegion);
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		eventBus.unregister(this);
```

Comment:
```suggestion
		eventBus.unregister(this);
		eventBus.unregister("regionchecker");
```

Just to make sure no leaks can occur